### ISSUE TITLE
how to use backward_passes_per_step in Tensorflow?

### ISSUE THREAD

**Issue by PANXiao1994** [horovod/horovod#1661]
**Environment:**
1. Framework: TensorFlow
2. Framework version: 1.15.0
3. Horovod version: unreleased version by using `HOROVOD_NCCL_HOME=/usr HOROVOD_GPU_BROADCAST=NCCL HOROVOD_WITH_TENSORFLOW=1 HOROVOD_GPU_ALLREDUCE=NCCL pip3 install git+https://github.com/horovod/horovod.git`
4. MPI version: Open MPI: 2.0.2
5. CUDA version: CUDA Version 10.0.130
6. NCCL version: libnccl-dev=2.4.8-1
7. Python version: python 3.5
8. OS and version: Debian GNU/Linux,   9 (stretch)
9. GCC version: gcc version 6.3.0 20170516 (Debian 6.3.0-18+deb9u1)

**Question:**
I want to implement gradient accumulation with the parameter `backward_passes_per_step`. What I have done is:
1. passed params in `hvd.DistributedOptimizer`:
```
hvd.DistributedOptimizer(
                self._optimizer, sparse_as_dense=True, backward_passes_per_step=self.params["update_cycle"], op=hvd.mpi_ops.Adasum)
```

2. I built the graph and defined `zero_op`(clear gradients), `collect_op`(accumulate gradients locally) and `train_op`(update params) as follows:
```
loss_var = tf.get_variable(name="replica_loss",
                           shape=[],
                           dtype=tf.float32,
                           initializer=tf.zeros_initializer,
                           trainable=False)
slot_variables = _replicate_variables(variables)
zero_variables_op = _zero_variables(slot_variables + [loss_var])
collect_grads_op = _collect_gradients(gradients, slot_variables)
collect_loss_op = tf.assign_add(loss_var, loss)
collect_op = tf.group(collect_loss_op, collect_grads_op, name="collect_op")
scale = 1.0 / float(update_cycle)
gradients = [(g + s) for (g, s) in zip(gradients, slot_variables)]
loss = scale * (loss + loss_var)
# clip gradients
if self.params["optimizer.clip_gradients"] > 0:
    gradients, _ = tf.clip_by_global_norm(
        gradients, self.params["optimizer.clip_gradients"])
    # grads_and_vars = _clip_gradients(grads_and_vars)
grads_and_vars = list(zip(gradients, variables))
# Create gradient updates.
grad_updates = self._optimizer.apply_gradients(
    grads_and_vars,
    global_step=tf.train.get_global_step(),
    name="train")
train_ops = {
    "zeros_op": zero_variables_op,
    "collect_op": collect_op,
    "train_op": grad_updates
}
``` 

3. I called N(equivalent to the value of  `backward_passes_per_step`) `collect_op` before calling one `train_op`:
```
while update_cycle[0] != update_cycle[1]:
    data = train_data.next()
    step_context.session.run(
        train_ops["collect_op"], feed_dict=data["feed_dict"])
    update_cycle[1] += 1
data = train_data.next()
update_cycle[1] = 1
return step_context.run_with_hooks(
    train_ops["train_op"], feed_dict=data["feed_dict"])
```

The code executed well, however, the performance degrades significantly: for wmt14 en2fr task, the resulting BLEU score is 11 (should be 40+ before this feature, the higher the better). I think this maybe caused by the `Adasum` operation when performing AllReduce, but I am not sure. 

I know that by using PyTorch, this will be much easier. But I need to implement this in TensorFlow. Is there any example like the one you offer in PyTorch of how to use the param `backward_passes_per_step` with `Adasum` operation in TensorFlow framework?
